### ISSUE TITLE
How many epochs were the official YOLOv8 pretrained models (detect, segment, pose, classify, obb) trained for?

### ISSUE THREAD

**Issue by SONA7459** [ultralytics/ultralytics#22182]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I have searched the Ultralytics GitHub issues and documentation, but I could not find an official statement about the number of epochs used to train the official YOLOv8 pretrained models.

Could you please clarify:
1. How many epochs were actually used to train the official YOLOv8 pretrained models for each task (detection, segmentation, pose estimation, classification, OBB)?
2. Do the number of training epochs differ between model sizes (n, s, m, l, x) or tasks?
3. Is there any official documentation or release note that specifies these training details for the pretrained weights?

### Additional

- I have seen some GitHub issues mentioning “500 epochs” for detection models, but I could not confirm this officially.
- The default config (default.yaml) shows 100 epochs, which might not reflect the actual training schedule of the released pretrained weights.

This clarification would be very valuable for researchers and practitioners working across different YOLOv8 tasks, especially for reproducing or fine-tuning the pretrained models.

Thank you very much!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @SONA7459, thank you for your interest in Ultralytics 🚀! This is an automated response to help you get started while an Ultralytics engineer reviews your question and will assist soon.

We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, and where many common questions may already be answered.

For this ❓ Question, could you please share any additional context that would help us provide the most accurate guidance?
- Which specific pretrained weights and tasks (detect/segment/pose/classify/obb) you’re focused on
- Any reproduction or fine-tuning goals you have (e.g., matching benchmarks, transfer learning setup)
- References to the configs or logs you’re comparing against

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by glenn-jocher** [comment]
Thanks for the thoughtful questions—there isn’t a single fixed epoch count published for YOLOv8; the released weights were trained to convergence with early stopping, so final epochs vary by task, model size, and recipe, and numbers like “500 epochs” are not an official standard. For context on models see the YOLOv8 overview and metrics at the [YOLOv8 model page](https://docs.ultralytics.com/models/yolov8/), and for how training is configured (including `epochs` and `patience`) see the [Train settings](https://docs.ultralytics.com/modes/train/); if you’re starting new work we recommend upgrading to [YOLO11](https://docs.ultralytics.com/models/yolo11/). For reproduction, start from the released weights and train longer with early stopping, for example:

```bash
yolo train model=yolov8s.pt data=your.yaml epochs=300 patience=50 imgsz=640
```

**Comment by SONA7459** [comment]
Hello Glenn,


Thank you very much for your detailed and thoughtful answer regarding the
training epochs of the YOLOv8 pretrained models.


Your explanation about early stopping and how the final number of epochs
can vary depending on the task, model size, and training recipe was very
helpful. The references you shared (YOLOv8 model page, training settings,
and YOLO11 recommendation) will be a great resource for my future work.


I truly appreciate your time and support in clarifying this important point.


Best regards,

Sona Sim


2025년 9월 23일 (화) 오전 8:24, Glenn Jocher ***@***.***>님이 작성:

**Comment by glenn-jocher** [comment]
Thanks, Sona—glad it helped; for reproducible knobs and defaults see the [Configuration guide](https://docs.ultralytics.com/usage/cfg/), practical guidance on choosing epochs and early stopping in the [Model training tips](https://docs.ultralytics.com/guides/model-training-tips/), and the exact `EarlyStopping(patience)` behavior in the [reference docs](https://docs.ultralytics.com/reference/utils/torch_utils/); if new questions come up, please open a new issue for visibility.
